components/intake.py

```diff
@@ -0,0 +1,42 @@
+import ctre
+import wpilib
+import enum
+
+
+class Action(enum.Enum):
+    Intake = 1
+    Outtake = -1
+    Stop = 0
+
+
+class Intake:
+    """Bunny intake"""
+    intake_motor = ctre.CANTalon
+    intake_pdp_channel = int
+    pdp = wpilib.PowerDistributionPanel
+    # Max current in amps to draw before stopping the motor
+    max_current = 6
+    release_bunny = False
+    holding_bunny = False
+
+    def spit_bunny(self):
+        """Spit out a bunny if one is being held"""
+        self.release_bunny = True
+
+    def execute(self):
+        if self.holding_bunny:
+            if self.release_bunny:
+                self.intake_motor.set(-0.6)
+                self.holding_bunny = False
+            else:
+                self.intake_motor.set(0.1)
+        else:
+            self.intake_motor.set(0.5)
+            if self.picked_up_bunny():
+                self.holding_bunny = True
+        self.release_bunny = False
+
+    def picked_up_bunny(self):
+        """Returns `True` if current draw indicates a bunny has been intaken"""
```

Comment:
intaken?

Reply:
The line was too long, I shortened it. Should it be intook or something? I'm think intaken is the correct tense...

Reply:
Could we inline this function?

Reply:
Yeah. Generally inlining into an if-else tree just harms readability though. Do you want me to do it?
```
if self.holding_bunny:
            if self.release_bunny:
                self.intake_motor.set(-0.6)
                self.holding_bunny = False
            else:
                self.intake_motor.set(0.1)
        else:
            self.intake_motor.set(0.5)
            if self.picked_up_bunny():
                self.holding_bunny = True
```
vs
```
if self.holding_bunny:
            if self.release_bunny:
                self.intake_motor.set(-0.6)
                self.holding_bunny = False
            else:
                self.intake_motor.set(0.1)
        else:
            self.intake_motor.set(0.5)
            if self.pdp.getCurrent(self.intake_pdp_channel) > self.max_current:
                self.holding_bunny = True
```

Reply:
Yeah I think so

Reply:
Sigh... it just hides the purpose with the implementation for the sake of 4 LOC.

Reply:
OK, you don't have to, but I just think it is more clear if it is inlined. Your choice